src/BloomExe/ImageProcessing/ImageUtils.cs

```diff
@@ -0,0 +1,162 @@
+﻿using System;
+using System.Diagnostics;
+using System.Drawing;
+using System.Drawing.Imaging;
+using System.Globalization;
+using System.IO;
+using System.Linq;
+using Palaso.UI.WindowsForms.ImageToolbox;
+using Logger = Palaso.Reporting.Logger;
+using TempFile = Palaso.IO.TempFile;
+
+namespace Bloom.ImageProcessing
+{
+	class ImageUtils
+	{
+		public static bool AppearsToBeJpeg(PalasoImage imageInfo)
+		{
+			/*
+			 * Note, each guid is VERY SIMILAR. The difference is only in the last 2 digits of the 1st group.
+			   Undefined  B96B3CA9
+				MemoryBMP  B96B3CAA
+				BMP    B96B3CAB
+				EMF    B96B3CAC
+				WMF    B96B3CAD
+				JPEG    B96B3CAE
+				PNG    B96B3CAF
+				GIF    B96B3CB0
+				TIFF    B96B3CB1
+				EXIF    B96B3CB2
+				Icon    B96B3CB5
+			 */
+			if(ImageFormat.Jpeg.Guid == imageInfo.Image.RawFormat.Guid)
+				return true;
+
+			if(ImageFormat.Jpeg.Equals(imageInfo.Image.PixelFormat))//review
+				return true;
+
+			if(string.IsNullOrEmpty(imageInfo.FileName))
+				return false;
+
+			return new[] { ".jpg", ".jpeg" }.Contains(Path.GetExtension(imageInfo.FileName).ToLower());
+		}
+
+		/// <summary>
+		/// Makes the image a png if it's not a jpg and saves in the book's folder.
+		/// If the image has a filename, replaces any file with the same name.
+		/// </summary>
+		/// <returns>The name of the file, now in the book's folder.</returns>
+		public static string ProcessAndSaveImageIntoFolder(PalasoImage imageInfo, string bookFolderPath)
+		{
+			LogMemoryUsage();
+
+			var isJpeg = AppearsToBeJpeg(imageInfo);
+			try
+			{
+				using (var image = new Bitmap(imageInfo.Image))
+					//nb: there are cases (undefined) where we get out of memory if we are not operating on a copy
+				{
+					//photographs don't work if you try to make the white transparent
+					if (!isJpeg && image is Bitmap)
```

Comment:
I don't see how image is Bitmap can be false...the previous line declares image to be a new Bitmap.